cmd/gardenlet/app/gardenlet.go

```diff
@@ -325,7 +325,7 @@ func NewGardenlet(ctx context.Context, cfg *config.GardenletConfiguration) (*Gar
 	}
 
 	seedClientMapBuilder := clientmapbuilder.NewSeedClientMapBuilder().
-		WithInCluster(cfg.SeedSelector == nil).
+		WithInCluster(true).
```

Comment:
Please remove `WithInCluster` and all the special handling related to `InCluster` in the client maps, as it's obsolete now.